tools/test/audio/comp_run.sh

```diff
@@ -121,18 +123,42 @@ run_testbench ()
 
 parse_args "$@"
 
-# Paths
-HOST_ROOT=../../testbench/build_testbench
-HOST_EXE=$HOST_ROOT/install/bin/testbench
-HOST_LIB=$HOST_ROOT/sof_ep/install/lib
-TPLG_LIB=$HOST_ROOT/sof_parser/install/lib
+# Path to topologies
 TPLG_DIR=../../build_tools/test/topology
 
+# Testbench path and executable
+if [[ -z $XTRUN ]]; then
+    TESTBENCH=../../testbench/build_testbench/install/bin/testbench
+else
+    BUILD_DIR=../../testbench/build_xt_testbench
+    TESTBENCH="$BUILD_DIR"/testbench
+    source "$BUILD_DIR"/xtrun_env.sh
+    XTRUN_CMD=$XTENSA_PATH/$XTRUN
+    if $VALGRIND; then
+	>&2 printf "WARNING: Ignoring VALGRIND with xt-run\n"
+	VALGRIND=false
+    fi
+fi
+
+if $VALGRIND; then
+    VALGRIND_CMD="valgrind --leak-check=yes --error-exitcode=1"
+else
+    VALGRIND_CMD=
+fi
+
+HOST_EXE="$XTRUN_CMD $TESTBENCH"
+
 # Use topology from component test topologies
 INFMT=s${BITS_IN}le
 OUTFMT=s${BITS_OUT}le
 TPLGFN=test-${DIRECTION}-ssp5-mclk-0-I2S-${COMP}-${INFMT}-${OUTFMT}-48k-24576k-codec.tplg
 TPLG=${TPLG_DIR}/${TPLGFN}
+[ -f "$TPLG" ] || {
+    echo
+    echo "Error: topology $TPLGFN does not exist."
```

Comment:
typo
```suggestion
    echo "Error: topology $TPLG does not exist."
```

Reply:
Fixed, the previous print was without path. Better this way.